llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -1451,10 +1451,12 @@ bool LoopVectorizationLegality::canVectorize(bool UseVPlanNativePath) {
   // Check whether the loop-related control flow in the loop nest is expected by
   // vectorizer.
   if (!canVectorizeLoopNestCFG(TheLoop, UseVPlanNativePath)) {
-    if (DoExtraAnalysis)
+    if (DoExtraAnalysis) {
+      LLVM_DEBUG(dbgs() << "LV: legality check failed: loop nest");
```

Comment:
unrelated to the original change?

Reply:
Yes, but doing it as good to have a debug message.

Reply:
Ok, generally it's preferable to keep things separate if possible. Fine to keep it here in the PR for now though